src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -472,12 +489,38 @@ void ReadFromRemote::addLazyPipe(Pipes & pipes, const ClusterProxy::SelectStream
                 max_remote_delay = std::max(try_result.delay, max_remote_delay);
         }
 
-        if (try_results.empty() || local_delay < max_remote_delay)
+        bool use_delayed_remote_source = false;
+        fiu_do_on(FailPoints::use_delayed_remote_source,
+        {
+            use_delayed_remote_source = true;
+        });
+
+        if (!use_delayed_remote_source)
         {
-            auto plan = createLocalPlan(
-                query, header, my_context, my_stage, my_shard.shard_info.shard_num, my_shard_count, my_shard.has_missing_objects);
+            const auto replicated_storage = std::dynamic_pointer_cast<StorageReplicatedMergeTree>(my_storage);
+            if (!replicated_storage)
+            {
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected lazy remote read from a non-replicated table: {}", my_storage->getName());
+            }
+            const UInt64 local_delay = replicated_storage->getAbsoluteDelay();
+            const UInt64 max_allowed_delay = current_settings[Setting::max_replica_delay_for_distributed_queries];
+
+            if (try_results.empty() && local_delay >= max_allowed_delay)
+            {
+                throw Exception(
+                    ErrorCodes::REPLICA_IS_STALE,
```

Comment:
Shouldn't we just use **existing** `ALL_REPLICAS_ARE_STALE`? Since in this case all replicas either stale or unavailable

Reply:
I considered using it, but what if the client just connected to the stale replica, which can't connect to other replicas?

Reply:
Well, AFAICS `ALL_REPLICAS_ARE_STALE` has the same issue

Reply:
Got it, changed the error to `ALL_REPLICAS_ARE_STALE `